internal/shader/syntax_test.go

```diff
@@ -1320,12 +1320,30 @@ func TestSyntaxOperatorShift(t *testing.T) {
 		stmt string
 		err  bool
 	}{
-		{stmt: "a := 1 << 2.0; _ = a", err: true},
-		{stmt: "a := 1.0 << 2; _ = a", err: true},
-		{stmt: "a := 1.0 << 2.0; _ = a", err: true},
+		// {stmt: "s := 1; var a float = float(1 << s); _ = a", err: true},
+		// {stmt: "s := 1; var a float = float(1.0 << s); _ = a", err: true},
+		// {stmt: "s := 1; var a int = int(1 << s); _ = a", err: false},
+		// {stmt: "s := 1; var a int = int(1.0 << s); _ = a", err: false},
+		// {stmt: "s := 1; a := 1 << s; _ = a", err: false},
+		// {stmt: "s := 1; a := 1.0 << s; _ = a", err: true},
+		// {stmt: "s := 1; a := int(1.0 << s); _ = a", err: false},
+		// {stmt: "s := 1; var a float = float(1.0 << s); _ = a", err: true},
+		// {stmt: "s := 1; var a float = 1.0 << s; _ = a", err: true},
+		// {stmt: "s := 1; var a float = 1 << s; _ = a", err: true},
+		// {stmt: "s := 1; var a int = 1.0 << s; _ = a", err: false},
+		// {stmt: "s := 1; var a int = 1 << s; _ = a", err: false},
```

Comment:
I assume this is still a work in progress. Please let me know if this is ready to review.